### PR TITLE
Integrate Standard Guidelines into Client

### PR DESCRIPTION
**Pull request by Cwagne17**
PR completes requirement # 1 of [Story 5026](https://app.clubhouse.io/clarkcan/story/5026/integrate-standard-guidelines-into-client). 

```
1. Add proxy to standard guidelines search routes in clark-gateway.
```

It adds the proxy to route to standard-guidelines-service
As well as adds complete documentation for the swagger docs

### CODE REVIEW

**Review by mfranz1 — APPROVED** [review]

File: src/modules/standard-guidelines/auth-module/AuthController.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+         *    description: authenticates a user 
+         *    tags:
+         *      - Standard Guideline Service
+         *    responses:
+         *      200:
+         *        description: OK
+         *      401:
+         *        description: INVALID ACCESS - User does not have a valid certificate
+         *      403:
+         *        description: FORBIDDEN - User is not whitelisted, please use a whitelisted certificate
```
Comment (EVogel1999) [inline src/modules/standard-guidelines/auth-module/AuthController.ts:24]: It doesn't throw these at all with this route, only with the routes that require authentication.  It returns an optional token (if they have the cert) and redirects them to the standard guidelines client regardless if they are authenticated or not.

File: src/modules/standard-guidelines/framework-module/FrameworkController.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+         *    parameters:
+         *      - in: path
+         *        name: id
+         *        schema:
+         *          type: string
+         *        required: true
+         *        description: The framework id
+         *    responses:
+         *      200:
+         *        description: OK
```
Comment (EVogel1999) [inline src/modules/standard-guidelines/framework-module/FrameworkController.ts:27]: Can we get a component schema for framework?

File: src/modules/standard-guidelines/guideline-module/GuidelineController.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+         *    parameters:
+         *      - in: path
+         *        name: id
+         *        schema:
+         *          type: string
+         *        required: true
+         *        description: The framework id
+         *    responses:
+         *      200:
+         *        description: OK
```
Comment (EVogel1999) [inline src/modules/standard-guidelines/guideline-module/GuidelineController.ts:27]: Can we also get a component schema for guidelines?

File: src/modules/standard-guidelines/guideline-module/GuidelineController.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+         *    parameters:
+         *      - in: path
+         *        name: id
+         *        schema:
+         *          type: string
+         *        required: true
+         *        description: The guideline id
+         *    responses:
+         *      200:
+         *        description: OK
```
Comment (EVogel1999) [inline src/modules/standard-guidelines/guideline-module/GuidelineController.ts:49]: Same here.

File: src/modules/standard-guidelines/search-module/SearchController.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+         *      - in: query
+         *        name: type
+         *        schema:
+         *          type: string
+         *        required: true
+         *        description: Determine whether to search for guidelines or frameworks, default is guidelines
+         *        example: guidelines
+         *    responses:
+         *      200:
+         *        description: OK
```
Comment (EVogel1999) [inline src/modules/standard-guidelines/search-module/SearchController.ts:63]: This should reference the framework component schema.

File: src/modules/standard-guidelines/search-module/SearchController.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
```
Comment (EVogel1999) [inline src/modules/standard-guidelines/search-module/SearchController.ts:119]: This should have a search index component schema in references.

**Review by EVogel1999 — CHANGES_REQUESTED** [review]
Looks good!  I just have one suggestion for the routes.  The 200s cases should reference a component schema if it returns a framework(s), guideline(s), or search index(es).

Edit: Make sure you whitelist GET /frameworks and GET /guidelines

**Review by pzalep1 — COMMENTED** [review]
Once Emily's comments are addressed this will be good to go.

**Review by jouypo23 — APPROVED** [review]

**Review by pzalep1 — APPROVED** [review]

**Review by EVogel1999 — APPROVED** [review]
Looks good!  Thank you for the schemas.

**Pull request merged**